policy-controller/k8s/index/src/outbound/index.rs

```diff
@@ -575,49 +635,59 @@ fn convert_backend(
     {
         Some(port) => port,
         None => {
-            return Some(Backend::Invalid {
+            return Some(Ok(Backend::Invalid {
                 weight: weight.into(),
                 message: format!("missing port for backend Service {name}"),
-            })
+            }))
         }
     };
     let service_ref = ServiceRef {
         name: name.clone(),
         namespace: backend.inner.namespace.unwrap_or_else(|| ns.to_string()),
     };
     if !services.contains_key(&service_ref) {
-        return Some(Backend::Invalid {
+        return Some(Ok(Backend::Invalid {
             weight: weight.into(),
             message: format!("Service not found {name}"),
-        });
+        }));
     }
 
+    let mut retry_policy = None;
     let filters = match filters
         .into_iter()
         .flatten()
-        .map(convert_gateway_filter)
+        .filter_map(|filter| convert_gateway_filter(filter, &mut retry_policy).transpose())
         .collect::<Result<_>>()
     {
         Ok(filters) => filters,
         Err(error) => {
-            return Some(Backend::Invalid {
+            return Some(Ok(Backend::Invalid {
                 weight: backend.weight.unwrap_or(1).into(),
                 message: format!("unsupported backend filter: {error}", error = error),
-            });
+            }));
         }
     };
 
-    Some(Backend::Service(WeightedService {
+    if retry_policy.is_some() {
+        return Some(Err(anyhow::anyhow!(
+            "HTTPRetryFilters may only be applied to HTTPRoute rules, not to inidividual backends"
+        )));
+    }
+
+    Some(Ok(Backend::Service(WeightedService {
         weight: weight.into(),
         authority: cluster.service_dns_authority(&service_ref.namespace, &name, port),
         name,
         namespace: ns.to_string(),
         port,
         filters,
-    }))
+    })))
 }
 
-fn convert_linkerd_filter(filter: api::httproute::HttpRouteFilter) -> Result<Filter> {
+fn convert_linkerd_filter(
+    filter: api::httproute::HttpRouteFilter,
+    retry_policy: &mut Option<LocalObjectReference>,
```

Comment:
returning a tuple would also work. if we returned a tuple, the code for checking if there was already a retry policy would need to be at the call site instead of in this function. i don't think it's a huge difference either way.

Reply:
@adleong what do you think of https://github.com/linkerd/linkerd2/pull/11117/commits/41f9099c7c57de6c28d075fe3292c737192bc031 ?

Reply:
Seems fine.  If we wanted to be super clean about this we could use `partition` to split the `rule.filters` into two iterators: one for filters that will get turned into retry policy and one for filters that get turned into filters in the proxy api.  But I'm splitting the bikeshed's hairs at this point.

Reply:
hmm, yeah, `partition` is a kind of nice idea, i might mess with this some more. but, the current solution seems fine if you're okay with it.